lib/configuration/hooks/grant/index.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+/**
+ * Module dependencies
+ */
+
+// Public node modules.
+const _ = require('lodash');
+const Grant = require('grant-koa');
+
+/**
+ * Grant hook
+ */
+
+module.exports = function (strapi) {
+  const hook = {
+
+    /**
+     * Default options
+     */
+
+    defaults: {
+      grant: {}
+    },
+
+    /**
+     * Initialize the hook
+     */
+
+    initialize: function (cb) {
+      const serverConf = {
+        server: {
+          protocol: strapi.config.ssl ? 'https' : 'http',
+          host: strapi.config.host + ':' + strapi.config.port
+        }
+      };
+      _.defaultsDeep(strapi.config.grant, serverConf);
```

Comment:
No need to use a constant for this, please write the object directly  as follow: 

```js
_.defaultsDeep(strapi.config.grant, {
  server: {
    protocol: strapi.config.ssl ? 'https' : 'http',
    host: strapi.config.host + ':' + strapi.config.port
  }
});
```